exporter/opentelemetry-exporter-prometheus/tests/test_prometheus_exporter.py

```diff
@@ -358,17 +358,18 @@ def test_target_info_enabled_by_default(self):
         counter.add(1)
         result = list(metric_reader._collector.collect())
 
-        for prometheus_metric in result[:0]:
-            self.assertEqual(type(prometheus_metric), InfoMetricFamily)
-            self.assertEqual(prometheus_metric.name, "target")
-            self.assertEqual(
-                prometheus_metric.documentation, "Target metadata"
-            )
-            self.assertTrue(len(prometheus_metric.samples) == 1)
-            self.assertEqual(prometheus_metric.samples[0].value, 1)
-            self.assertTrue(len(prometheus_metric.samples[0].labels) == 2)
-            self.assertEqual(prometheus_metric.samples[0].labels["os"], "Unix")
-            self.assertEqual(prometheus_metric.samples[0].labels["histo"], "1")
+        self.assertEqual(len(result), 2)
+
+        prometheus_metric = result[0]
+
+        self.assertEqual(type(prometheus_metric), InfoMetricFamily)
+        self.assertEqual(prometheus_metric.name, "target")
+        self.assertEqual(prometheus_metric.documentation, "Target metadata")
+        self.assertTrue(len(prometheus_metric.samples) == 1)
+        self.assertEqual(prometheus_metric.samples[0].value, 1)
+        self.assertTrue(len(prometheus_metric.samples[0].labels) == 2)
+        self.assertEqual(prometheus_metric.samples[0].labels["os"], "Unix")
+        self.assertEqual(prometheus_metric.samples[0].labels["histo"], 1)
```

Comment:
This label had the value of a string before `"1"`, but it is an integer `1`.
Thus, I changed it to an integer.

Reply:
Shouldn't it be string as Prometheus format only supports string key-value for labels?

Reply:
Yes you are right, this raises also an exception `AttributeError: ("'int' object has no attribute 'replace'", Metric(target, Target metadata, info` (`prometheus_client\openmetrics\exposition.py", line 33, in <listcomp> k, v.replace('\\', r'\\').replace('\n', r'\n').replace('"', r'\"'))`) in this minimal example:

```python
import time

from prometheus_client import start_http_server

from opentelemetry import metrics
from opentelemetry.exporter.prometheus import PrometheusMetricReader
from opentelemetry.sdk.metrics import MeterProvider
from opentelemetry.sdk.resources import SERVICE_NAME, Resource

resource = Resource(attributes={
    SERVICE_NAME: "your-service-name",
    "foo": "bar",
    "hist": 1,
})

start_http_server(port=8000, addr="localhost")
reader = PrometheusMetricReader()
provider = MeterProvider(resource=resource, metric_readers=[reader])
metrics.set_meter_provider(provider)

meter = metrics.get_meter("my.meter.name")

work_counter = meter.create_counter(
    "work.counter", unit="1", description="Counts the amount of work done"
)

while True:
    work_counter.add(1, {"work.type": "Debug"})
    print("doing some work...")
    time.sleep(1)
```

I will write a bug report and will directly address the bug in this PR.